src/hardware/vga_draw.cpp

```diff
@@ -904,12 +905,12 @@ static void VGA_ProcessSplit()
 	vga.draw.address_line = 0;
 }
 
-static uint16_t from_rgb_888_to_565(const uint32_t rgb888)
+static constexpr uint16_t to_rgb_565(const Bgra8888 bgra_8888)
 {
-	const auto rgb565 = ((rgb888 & 0xf80000) >> 8) +
-	                    ((rgb888 & 0xfc00) >> 5) + ((rgb888 & 0xf8) >> 3);
-
-	return check_cast<uint16_t>(rgb565);
+	const auto rgb_565 = Rgb565(bgra_8888.GetRed8(),
+	                            bgra_8888.GetGreen8(),
+	                            bgra_8888.GetBlue8());
+	return rgb_565.pixel;
```

Comment:
Can we move this into `Rgb565` or `Rgba8888`?

Reply:
I'd rather the Bgra8888 and Rgb565's classes not have other format code it, and vice-versa. We need some common bucket where these multi-format conversion calls can live.

But because this is the only place in the code we convert from Bgra8888 to Rgb565, I figured it was fine.

Reply:
I can eliminate the function and directly move between formats given because both classes can read and write from their r8, g8, and b8 channels (the designs are nicely lining up!)